tests/core/test_stellargraph.py

```diff
@@ -427,3 +428,44 @@ def test_feature_conversion_from_iterator():
     ab = gs.get_feature_for_nodes([4, 5], "B")
     assert ab.shape == (2, 10)
     assert ab[:, 0] == pytest.approx([4, 5])
+
+
+def example_benchmark_graph(feature_size=None, n_nodes=100, n_edges=200, n_types=4):
+    G = nx.Graph()
+
+    G.add_nodes_from(range(n_nodes))
+    edges = [
+        (random.randint(0, n_nodes - 1), random.randint(0, n_nodes - 1))
+        for _ in range(n_edges)
+    ]
+    G.add_edges_from(edges)
+
+    # Add example features
+    if feature_size is not None:
+        for v in G.nodes():
+            G.nodes[v]["feature"] = np.ones(feature_size)
+            G.nodes[v]["label"] = v % n_types
+
+        G = StellarGraph(G, node_features="feature")
+    else:
+        G = StellarGraph(G)
+    return G
+
+
+@pytest.mark.benchmark(group="StellarGraph creation", timer=snapshot)
+# various element counts, to give an indication of the relationship
+# between those and memory use (0,0 gives the overhead of the
+# StellarGraph object itself, without any data)
+@pytest.mark.parametrize("num_nodes,num_edges", [(0, 0), (100, 0), (100, 200)])
+# various feature sizes (including no features) to capture that cost
+@pytest.mark.parametrize("feature_size", [None, 1, 100])
+def test_benchmark_creation(allocation_benchmark, feature_size, num_nodes, num_edges):
+    g = example_benchmark_graph(feature_size, num_nodes, num_edges)
```

Comment:
Isn't `g` here already a StellarGraph? Why does `f()` wrap it again? Shouldn't `f()` just return `g`?

Reply:
No, the benchmarked function `f` needs to do any allocations inside it for them to be measured correctly. This is written a bit strangely while `StellarGraph` inherits from `nx.Graph`, because there's no way to refer to a `StellarGraph` and its data as purely a networkx object. Once #501 lands, and, supposing we had a `.to_networkx` method, I'd rewrite this as is as:

```python
g = example_benchmark_graph(...)
nx_g = g.to_networkx()

def f():
    if feature_size is None:
        return StellarGraph(nx_g)
    else:
        ...
```

Your comment makes me realise:

- the name of the function could be improved: I'll rename it to `test_benchmark_creation_from_networkx`. 
- maybe `example_benchmark_graph` should be changed to return the networkx graphs directly

I'll also try to add a benchmark that uses the new style `node_features` method to specify features, but this benchmark is enough to validate the `alloc.py` approach I've gone with.